### PR TITLE
Add s3 to user

### PR DESCRIPTION
**Pull request by jonmagic**
This basically just adds a `PhotoStore` model which stores S3 creds and uses the fog gem to connect to S3. Technically a `User` can have many photo stores but for now the UI limits you to one.

foto

No tests on this yet, but I'll add integration tests for the UI (or at least the controller) and for the PhotoStore directory and connection methods before merging this PR.

/cc @johndbritton

### CODE REVIEW

**Comment by johndbritton** [comment]
Looks good. Do you mind me dropping line comments in to ask style / design type questions?

**Comment by jonmagic** [comment]
@johndbritton go for it :)

File: Gemfile [hunk 1]
```diff
@@ -15,6 +16,7 @@ gem "rolify"
 gem "sass-rails"
 gem "turbolinks"
 gem "uglifier"
+gem "unf"
```
Comment (johndbritton) [inline Gemfile:19]: What exactly does `unf` do? I didn't really understand the [readme](https://github.com/knu/ruby-unf/blob/master/README.md).

Comment (jonmagic) [inline Gemfile:19]: All I know is the aws (or fog) gem was pissed off without it. After adding it warning messages went away and aws requests seemed to speed up quite a bit.

**Comment by jonmagic** [comment]
@johndbritton this PR is probably doing more than it should, I think I should probably pull out the test harness updates to a second PR, what do you think?

**Comment by johndbritton** [comment]
Sounds good to me.

**Comment by jonmagic** [comment]
@johndbritton ok, I think I'm done with this PR now :) I'm not terribly proud of the commit history, but the content of the PR is where I want it to be.

**Pull request merged**